### PR TITLE
#169500769 Generate starter migrations and models for the API

### PR DESCRIPTION
**Pull request by tech-lady-zz**
#### What does this PR do?
- This PR sets up the database, models, migrations, associations, integration tests using test driven development
#### Description of Task to be completed?
- Run migrations for the API and generate models for articles, comments and users. Include roles and associations.
#### How should this be manually tested?
- run bundle
- run rails db:setup
- run rails db:migrate
- run rspec spec
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/169500769

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,8 @@ source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.5.5'
+gem 'bcrypt'
+gem 'dotenv-rails', groups: [:development, :test]
```
Comment (ChigboIO) [inline Gemfile]: Not a big deal, but you have a gem group of `:development` and `:test` in line [32](https://github.com/tech-lady/Mediaum-api/pull/3/files#diff-8b7db4d5cc4b8f6dc8feb7030baa2478R32), why not move this gem within the block and remove the "group" option

File: app/models/article.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Article < ApplicationRecord
+  has_many :responses
```
Comment (ChigboIO) [inline app/models/article.rb]: It's best practice to always define your dependent action in your `has_many` associations. This defines what happens to the associated records when the parent record is destroyed. This line for instance could be written as:
```ruby
has_many :responses, dependent: :destroy
```
...which means you want to associated responses to be destroyed when the parent Article is destroyed.
Check [here](https://guides.rubyonrails.org/association_basics.html#dependent), [here](https://apidock.com/rails/ActiveRecord/Associations/ClassMethods/has_many) and [here](https://makandracards.com/makandra/32175-don-t-forget-automatically-remove-join-records-on-has_many-through-associations) for more information and other options that could be passed to  the `dependent` key.

Comment (ChigboIO) [reply app/models/article.rb]: The same thing goes for all your `has_many`'s

File: db/migrate/20191111090601_create_articles.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateArticles < ActiveRecord::Migration[6.0]
+  def change
+    create_table :articles do |t|
+      t.string :title
+      t.string :content
+      t.integer :user_id
```
Comment (ChigboIO) [inline db/migrate/20191111090601_create_articles.rb:6]: Prefer using references to create your foreign keys and add the `foreign_key: true` option. This line could instead be:
```ruby
t.references :user, foreign_key: true
```
The `references` will automatically create index in this field which is necessary for fast lookup. And the `foreign_key: true` option sets the field up as a foreign_key to the users table and applies foreign key contraints.

File: db/migrate/20191111092601_create_bookmarks.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateBookmarks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :bookmarks do |t|
+      t.integer :user_id
+      t.integer :article_id
```
Comment (ChigboIO) [inline db/migrate/20191111092601_create_bookmarks.rb:5]: The same thing about using t.references goes for user_id and article_id here. Reference: https://stackoverflow.com/a/39502978/5699251

File: spec/factories/users.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :user do
+    name { "Hollaryem" }
+    username { "Yemyem" }
+    email { "yemyem@yahoo.com" }
+    password { "1234_@abcd" }
```
Comment (ChigboIO) [inline spec/factories/users.rb]: Not super necessary, but I like using [faker](https://github.com/faker-ruby/faker) in factories to randomly generate different values for each field

**Review by ChigboIO — COMMENTED** [review]
Good work with the model setup with tests 👍

**Comment by tech-lady-zz** [comment]
Thanks for the feedback @ChigboIO. All feedback have been addressed

**Comment by ChigboIO** [comment]
Awesome! Nice work, @tech-lady 👍

**Review by raphaelakpan — APPROVED** [review]
Looks good 🎉

**Comment by tech-lady-zz** [comment]
Thanks for the approval @raphaelakpan

**Pull request merged**